plugin/src/test/groovy/org/gradle/testretry/testframework/SpockFuncTest.groovy

```diff
@@ -647,6 +648,130 @@ class SpockFuncTest extends AbstractPluginFuncTest {
         gradleVersion << GRADLE_VERSIONS_UNDER_TEST
     }
 
+    @Unroll
+    def "build fails if a param test has failed once but never passed (gradle version #gradleVersion)"() {
+        given:
+        buildFile << """
+            test.retry.maxRetries = 3
+            test.retry.failOnPassedAfterRetry = false
+        """
+
+        writeTestSource """
+            package acme
+            import java.nio.file.Paths
+            import java.nio.file.Files
+
+            @spock.lang.Stepwise
+            class StepwiseTests extends spock.lang.Specification {
+                def "parentTest"() {
+                    expect:
+                    true
+                }
+
+                @spock.lang.IgnoreIf({Files.exists(Paths.get("build/marker.file")) })
+                def "childTest"() {
```

Comment:
❓ Why "child" and "parent"?

Reply:
The parent is indeed not required here anymore.